### PR TITLE
Fix login

### LINKED ISSUE #43: Error occur ,after enter correct credentials
When i run hsc command after installation  , it will asked for credentials to enter. I enter the correct details but it showing error.

Error:-

Traceback (most recent call last):
  File "C:\Python27\Scripts\hsc-script.py", line 11, in <module>
    load_entry_point('hsc==1.2.1', 'console_scripts', 'hsc')()
  File "c:\python27\lib\site-packages\hsc\crawler.py", line 221, in main
    if not crawler.authenticate():
  File "c:\python27\lib\site-packages\hsc\crawler.py", line 62, in authenticate
    return self.login(username, password)
  File "c:\python27\lib\site-packages\hsc\crawler.py", line 46, in login
    self.get_number_of_submissions()
  File "c:\python27\lib\site-packages\hsc\crawler.py", line 68, in get_number_of_submissions
    self.total_submissions = resp.json()['total']
  File "c:\python27\lib\site-packages\requests\models.py", line 898, in json
    return complexjson.loads(self.text, **kwargs)
  File "c:\python27\lib\json\__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "c:\python27\lib\json\decoder.py", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "c:\python27\lib\json\decoder.py", line 382, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded

I also use the direct login command hsc -l 34 -p testpassword -u testuser
But this also not working!

Help me to resolve it!

### PR DESCRIPTION
**Pull request by rishabhsingh971**
- Add user agent in requests, closes #43 
- Use post instead of get in login request
- Remove extra whitespaces

### CODE REVIEW

File: hsc/crawler.py [hunk 1]
```diff
@@ -40,11 +41,13 @@ def __init__(self):
...
-		resp = self.session.get(self.login_url, auth=(username, password))
-		self.cookies = self.session.cookies.get_dict()
-		self.headers = resp.request.headers
-		self.get_number_of_submissions()
-		return self.total_submissions != 0
+		resp = self.session.post(self.login_url, auth=(username, password), headers={'user-agent': self.user_agent})
+		data = resp.json()
+		if data['status']:
+			self.cookies = self.session.cookies.get_dict()
+			self.headers = resp.request.headers
```
Comment (rajatgoyal715) [inline hsc/crawler.py:48]: @rishabhsingh971 Do you know the reason for setting cookies and headers here?

File: hsc/crawler.py [hunk 2]
```diff
@@ -40,11 +41,13 @@ def __init__(self):
...
+			self.get_number_of_submissions()
+		return data['status']
```
Comment (rajatgoyal715) [inline hsc/crawler.py:50]: Earlier it was `return self.total_submissions != 0`. Will this change to `return data['status']` have some impact?

**Review by rajatgoyal715 — CHANGES_REQUESTED** [review]
@rishabhsingh971 I've added you as a contributor to the test pypi (https://test.pypi.org/project/hsc/). Please upload the screenshots here once you test the changes.

File: hsc/crawler.py (hunk 1 above)
Comment (rishabhsingh971) [reply hsc/crawler.py:48]: Yes, It's for authentication. We set the response cookies and headers sent by server when successfully logged in and then send them in further requests (as done by browsers)

File: hsc/crawler.py (hunk 2 above)
Comment (rishabhsingh971) [reply hsc/crawler.py:50]: Yes because a user may have 0 submissions. So it's wrong (hack) to judge login based on the count.

**Comment by rishabhsingh971** [comment]
image

**Review by rajatgoyal715 — APPROVED** [review]

**Pull request merged**